src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenForEachTests.cs

```diff
@@ -5327,5 +5327,254 @@ public static class Extensions
 }";
             CompileAndVerify(source, parseOptions: TestOptions.Regular9, expectedOutput: "123123");
         }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/73741")]
+        public void MutatingThroughRefFields_01(
+            [CombinatorialValues("ref", "")] string eRef,
+            [CombinatorialValues("readonly", "")] string vReadonly)
+        {
```

Comment:
Consider adding a smoke test for LINQ iteration variables and `using` variables. They have the same "can't assign" semantics as `foreach` iteration variables and hence are possibly subject to the same issue.

Reply:
There are tests for `using` variables added in this PR in CodeGenUsingStatementTests.cs.
LINQ iteration variables don't seem to be affected by this bug (the iteration variable is not a LocalSymbol so the BoundFieldAccess has always IsByValue set to false), but I will add similar tests.